v0/destinations/airship/transform.js

```diff
@@ -0,0 +1,307 @@
+  }
+
+  tagPayload.add = { rudderstack_integration: [] };
+  tagPayload.remove = { rudderstack_integration: [] };
+  let timestamp = getFieldValueFromMessage(message, "timestamp");
+  timestamp = new Date(timestamp).toISOString().replace(/\.[0-9]{3}/, "");
+
+  attributePayload.attributes = [];
+  Object.keys(traits).forEach(key => {
+    if (typeof traits[key] === "boolean") {
+      const tag = key.toLowerCase().replace(/\./g, "_");
+      if (traits[key] === true) {
+        tagPayload.add.rudderstack_integration.push(tag);
+      }
+      if (traits[key] === false) {
+        tagPayload.remove.rudderstack_integration.push(tag);
+      }
+    }
+    if (typeof traits[key] !== "boolean") {
+      const attribute = {};
+      attribute.action = "set";
+      const keyMapped = RESERVED_TRAITS_MAPPING[key.toLowerCase()];
+      if (keyMapped) {
+        attribute.key = keyMapped;
+      } else {
+        attribute.key = key.replace(/\./g, "_");
+      }
+      attribute.value = traits[key];
+      attribute.timestamp = timestamp;
+      attributePayload.attributes.push(attribute);
+    }
+  });
+
+  let tagResponse;
+  let attributeResponse;
+  const arrayPayload = [];
+  if (
+    tagPayload.add.rudderstack_integration.length ||
+    tagPayload.remove.rudderstack_integration.length
+  ) {
+    if (!tagPayload.add.rudderstack_integration.length) delete tagPayload.add;
+    if (!tagPayload.remove.rudderstack_integration.length)
+      delete tagPayload.remove;
+    tagResponse = defaultRequestConfig();
+    tagResponse.endpoint = `${BASE_URL}/api/named_users/tags`;
+    tagResponse.headers = {
+      "Content-Type": "application/json",
+      Accept: "application/vnd.urbanairship+json; version=3",
+      Authorization: `Basic ${btoa(`${appKey}:${appSecret}`)}`
+    };
+    tagResponse.method = defaultPostRequestConfig.requestMethod;
+    tagResponse.body.JSON = removeUndefinedAndNullValues(tagPayload);
+    arrayPayload.push(tagResponse);
+  }
+  if (attributePayload.attributes.length) {
+    attributeResponse = defaultRequestConfig();
+    attributeResponse.endpoint = `${BASE_URL}/api/named_users/${getFieldValueFromMessage(
+      message,
+      "userId"
+    )}/attributes`;
+    attributeResponse.headers = {
+      "Content-Type": "application/json",
+      Accept: "application/vnd.urbanairship+json; version=3",
+      Authorization: `Basic ${btoa(`${appKey}:${appSecret}`)}`
+    };
+    attributeResponse.method = defaultPostRequestConfig.requestMethod;
+    attributeResponse.body.JSON = removeUndefinedAndNullValues(
+      attributePayload
+    );
+    arrayPayload.push(attributeResponse);
+  }
+
+  return arrayPayload;
+};
+
+const trackResponseBuilder = async (message, { Config }) => {
+  let name = message.event;
+  if (!name) {
+    throw new CustomError("event name is required for track", 400);
+  }
+
+  name = name.toLowerCase();
+  const payload = constructPayload(message, trackMapping);
+  payload.name = name.replace(" ", "_");
```

Comment:
```suggestion
  payload.name = name.replace(/\s+/g, "_");
```
Handling case where multiple space is provided